plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/listeners/ConfigureDependentFrameworksListener.java

```diff
@@ -0,0 +1,212 @@
+		IDockerConnection[] cons = mgr.getConnections();
+		String httpHost = dockerHost.replace("tcp://", "http://");
+		for( int i = 0; i < cons.length; i++ ) {
+			System.out.println(dockerHost + " vs " + cons[i].getUri());
+			if( cons[i].getUri().equals(dockerHost) || cons[i].getUri().equals(httpHost)) {
+				return true;
+			}
+		}
+		return false;
+	}
+	private void configureDocker(IServer server, ADBInfo adb) {
+		try {
+			DockerConnectionManager mgr = org.eclipse.linuxtools.docker.core.DockerConnectionManager.getInstance();
+			final String dockerHost = adb.env.get("DOCKER_HOST");
+			if( !dockerConnectionExists(mgr, dockerHost)) {
+				final Builder tcpConnectionBuilder = new DockerConnection.Builder()
+						.name(getNextName(server, mgr)).tcpHost(dockerHost);
+				String tlsVerifyString = adb.env.get("DOCKER_TLS_VERIFY");
+				boolean tlsVerify = (Integer.parseInt(tlsVerifyString) != 0);
+				// For some reason, the connection fails if i set this. 
+				// Not sure why the disconnect between what adbinfo tells me to use and what works upstream
+//				if( tlsVerify ) {
+//					String tlsCertPath = adb.env.get("DOCKER_CERT_PATH");
+//					tcpConnectionBuilder.tcpCertPath(tlsCertPath);
+//				}
+				DockerConnection con = tcpConnectionBuilder.build();
+				IDockerConnection[] other = mgr.getConnections();
+				mgr.addConnection(con);
+			}
+		} catch(DockerException de) {
+			de.printStackTrace();
+		}
+	}
+	private void configureOpenshift(ADBInfo adb) {
+		Collection<IConnection> connections = ConnectionsRegistrySingleton.getInstance().getAll();
+		Iterator<IConnection> it = connections.iterator();
+		String soughtHost = adb.openshiftHost + ":" + adb.openshiftPort;
+		boolean found = false;
+		while(it.hasNext() && !found) {
+			IConnection c = it.next();
+			if( c.getType() == ConnectionType.Kubernetes) {
+				String host = c.getHost();
+				if( host.equals(soughtHost)) {
+					found = true;
+				}
+			}
+		}
+		
+		if( !found ) {
+			ConnectionsFactoryTracker connectionsFactory = new ConnectionsFactoryTracker();
+			connectionsFactory.open();
+			IConnectionFactory factory = connectionsFactory.getById(IConnectionsFactory.CONNECTIONFACTORY_OPENSHIFT_ID);
+			IConnection con = factory.create(soughtHost);
+			((Connection)con).setAuthScheme("Basic");
+			((Connection)con).setUsername("test-admin");
+			((Connection)con).setPassword("password");
+			ConnectionsRegistrySingleton.getInstance().add(con);
+		}
+	}
+	
+	private ADBInfo loadADBInfo(IServer server) {
+		try {
+			ILaunchConfigurationWorkingCopy wc = new CDKLaunchConfigUtility().
+					createExternalToolsLaunchConfig(server, "adbinfo", server.getName() + " adbinfo");
+			ILaunch launch2 = wc.launch("run", new NullProgressMonitor());
+			final IProcess[] processes = launch2.getProcesses();
+			IStreamMonitor mon = processes[0].getStreamsProxy().getOutputStreamMonitor();
+			final ArrayList<String> lines = new ArrayList<String>();
+			mon.addListener(new IStreamListener() {
```

Comment:
Is it guaranteed that text is always appended in complete lines? Maybe it makes sense to check if appended text ends with line separator, and also split text to lines to add all true lines into the array at this moment. If text does not end with line separator, then the last line is not complete, it should be saved into a temporary string and concatenated with first line in the next contribution.
Or, to make things simple and safe, all text contributions may be added to StringBuilder rather than to List.